MetaMask/metamask-extension
# Bug 17630 silent ledger failures

## Explanation

Issue explained via story:

1. User initiates a transaction with an unlocked Ledger
2. Confirmation Modal pops up, and no warning is present because the Ledger is unlocked
3. Ledger locks while the Confirmation is open (e.g. the user may be hesitating the confirmation for a bit, or may be playing around with fees etc.)
4. User hits "Confirm" and the transaction fails silently as the Ledger is now locked
5. Issue is applicable to Confirmation modals for transactions, signing typed data, signing messages etc.

## Screenshots/Screencaps

### Before

When Ledger device enters locked state, the transaction is allowed to proceed:

![image](https://user-images.githubusercontent.com/82925669/218348794-bd154304-af87-4338-a0bc-e0a75b40ae5f.png)

Confirming transaction results in a failed transaction with vague error (code: `0x6b0c`):

![image](https://user-images.githubusercontent.com/82925669/218348912-51334aa4-b287-4e6d-9604-ed1f36076b24.png)

### After

When Ledger device enters lock state, the UI responds with the following and transaction cannot proceed:

![image](https://user-images.githubusercontent.com/82925669/219546676-12ec798b-87ef-4884-aa9b-b9a46e5a8aed.png)

Once the Ledger device is unlocked, the UI returns to normal and the transaction can proceed:

![image](https://user-images.githubusercontent.com/82925669/219546719-3ab4d475-3f10-4332-9588-02506882c2fb.png)


## Manual Testing Steps

1. Initiate transaction with Ledger
2. While on the confirm screen, wait for the Ledger to lock. Transaction should be blocked with `ledgerLocked` warning message showing
3. Unlock Ledger. The `ledgerLocked` warning should no longer be present and the transaction should now be able to proceed
4. Confirm transaction and proceed as normal

## Pre-merge author checklist


## Pre-merge reviewer checklist


If further QA is required (e.g. new feature, complex testing steps, large refactor), add the `Extension QA Board` label.

In this case, a QA Engineer approval will be be required.
